In [1]:
from math import factorial
from math import exp

In [2]:
def bernoulli(k, n, p):
    return (factorial(n)/(factorial(k)*factorial(n-k))) *p**k * (1-p)**(n-k)

In [3]:
def poisson(p, n, m):
    l = p*n
    return l**m/factorial(m) * exp(-l)

In [4]:
def combinations(n, k):
    return int(factorial(n) / (factorial(k) * factorial(n - k)))

### 1. Вероятность того, что стрелок попадет в мишень, выстрелив один раз, равна 0.8. Стрелок выстрелил 100 раз. Найдите вероятность того, что стрелок попадет в цель ровно 85 раз.

$p (k=85) = C_{100}^{85}p^{85}q^{15} = \frac{100!}{85!\cdot 15!} \cdot 0.8^{85}\cdot0.2^{15}$

In [5]:
bernoulli(85, 100, 0.8)  #примерно 4.8%

0.048061793700746355

### 2. Вероятность того, что лампочка перегорит в течение первого дня эксплуатации, равна 0.0004. В жилом комплексе после ремонта в один день включили 5000 новых лампочек. Какова вероятность, что ни одна из них не перегорит в первый день? Какова вероятность, что перегорят ровно две?

1) ни одна не перегорит:
m = 0, n = 5000.

$\lambda = 0.0004 \cdot 5000 = 2$

In [6]:
# вероятность, что ни одна не перегорит:
poisson(0.0004, 5000, 0) # примерно 13.5%

0.1353352832366127

In [7]:
# вероятность, что 2 перегорят:
poisson(0.0004, 5000, 2) # примерно 27%

0.2706705664732254

### 3. Монету подбросили 144 раза. Какова вероятность, что орел выпадет ровно 70 раз?

In [10]:
bernoulli(70, 144, 0.5) # примерно 6.3%

0.06281178035144776

### 4. В первом ящике находится 10 мячей, из которых 7 - белые. Во втором ящике - 11 мячей, из которых 9 белых. Из каждого ящика вытаскивают случайным образом по два мяча. Какова вероятность того, что все мячи белые? Какова вероятность того, что ровно два мяча белые? Какова вероятность того, что хотя бы один мяч белый?

In [28]:
# вероятность достать 2 белых мяча из 1го ящика
all_white1=combinations(7, 2)/ combinations(10,2)
all_white1

0.4666666666666667

In [29]:
# вероятность достать 2 мяча другого цвета из 1го ящика
not_white1=combinations(3, 2)/ combinations(10,2)
not_white1

0.06666666666666667

In [56]:
# 1 белый и 1 не белый из 1 ящика
one_white1=combinations(3, 1) * combinations(7, 1) / combinations(10, 2)
one_white1

0.4666666666666667

In [33]:
# вероятность достать 2 белых мяча из 2го ящика
all_white2=combinations(9, 2)/ combinations(11,2)
all_white2

0.6545454545454545

In [34]:
# вероятность достать 2 мяча другого цвета из 2го ящика
not_white2=combinations(2, 2)/ combinations(11,2)
not_white2

0.01818181818181818

In [43]:
# 1 белый и 1 не белый из 2го ящика
one_white2=combinations(2, 1)* combinations(9,1) /  combinations(11,2)
one_white2

0.32727272727272727

In [44]:
# все мячи белые:
all_white1 * all_white2

0.3054545454545455

In [65]:
# строго 2 мяча белые
all_white1 * not_white2 + not_white1 * all_white2 + one_white1 * one_white2

0.20484848484848486

In [66]:
# хотя бы 1 белый
1 - not_white1* not_white2

0.9987878787878788

Какова вероятность того, что все мячи белые? - примерно 30.55%

Какова вероятность того, что ровно два мяча белые? - примерно 20.48%

Какова вероятность того, что хотя бы один мяч белый? - примерно 99.88%